src/types.ts

```diff
@@ -2,18 +2,23 @@ import { DataSourceJsonData } from '@grafana/data';
 import { DataQuery } from '@grafana/schema';
 
 /**
- * Query used with Cribl Search
+ * Query used with Cribl Search.  Can either use a saved search or run an adhoc query.
  */
-export interface CriblQuery extends DataQuery {
-  /**
-   * ID of the Cribl saved search
-   */
-  savedSearchId: string;
-  /**
-   * Max number of results to fetch
-   */
-  maxResults: number;
-}
+export type CriblQuery = DataQuery & (
+  {
+    type: 'adhoc';
+    /**
+     * Ad-hoc query (Kusto)
+     */
+    query: string;
+  } | {
```

Comment:
Didn't know you could do this with my OOP mindset,I had to read up on discriminated unions after seeing this. I'm liking Typescript more and more to say the least!